ral-facilities/datagateway-api
# Fix Pip Install Issues

This PR will close #467

## Description
These changes allow DataGateway API to be installed using `pip install` - the fix is simply just to explicitly state which version of Flask should be used instead of the installation process trying to figure that out. Normally that wouldn't be an issue, but we're having to use quite an old version of Flask that still supports Python 3.6. There's also some updates to `poetry.lock` as a result - a lot of that is just upgrading other dependencies as the file hasn't changed in a little while.

I've added a job to the CI to keep track of this - you can see in [this run](https://github.com/ral-facilities/datagateway-api/actions/runs/7289367196) (before Flask was pinned), that new job fails on Python 3.8+, giving the same error that was originally found but the CI now passes.

When the PR is merged, it will create v9.0.2 so you'll be able to use this new version with SciGateway Ansible.

## Testing Instructions
Take a look at the CI and check you're happy with it and try to pip install the API on your own machine.


## Agile Board Tracking
Connect to #467
